flexmeasures/data/models/planning/tests/conftest.py

```diff
@@ -219,6 +219,39 @@ def process(db, building, setup_sources) -> dict[str, Sensor]:
     return _process
 
 
+@pytest.fixture(scope="module")
+def add_usage_forecast(db, add_battery_assets, setup_sources) -> Sensor:
+    """
+    Set up a constant usage forecast
+    """
+
+    battery = add_battery_assets["Test battery"]
+
+    # 1 days of test data
+    time_slots = initialize_index(
+        start=pd.Timestamp("2015-01-01").tz_localize("Europe/Amsterdam"),
+        end=pd.Timestamp("2015-01-02").tz_localize("Europe/Amsterdam"),
+        resolution="15T",
+    )
+
+    usage_forecast_sensor = Sensor(
+        name="usage forecast",
+        unit="MWh",
+        event_resolution=timedelta(minutes=15),
+        generic_asset=battery,
+    )
+    db.session.add(usage_forecast_sensor)
+    db.session.flush()
+
+    usage_forecast = [-battery.get_attribute("capacity_in_mw")] * len(time_slots)
```

Comment:
Confusing step that may be a mistake, or otherwise could at least use a comment. You're setting a loss of 2 MWh/15min, i.e. a loss rate of 8 MW. This might explain why the battery decides to charge at its max capacity, trying to make up for the loss before completely draining (but, as you already commented, we actually expect it to fail when the SoC drops below 0).

Reply:
I chose that value because I wanted to see If the battery could charge at a higher rate (8 MW) than it should (2MW). This is related to the discussion https://github.com/FlexMeasures/flexmeasures/discussions/692. Perhaps I'm not understanding well but I think the derivative max, derivative min should take the resolution into account:

https://github.com/FlexMeasures/flexmeasures/blob/72f65ecb38a304a2a164d96172cfda4767ca0fe2/flexmeasures/data/models/planning/storage.py#L200-L209